frontend/src/components/ExplorerCard/ExplorerCard.tsx

```diff
@@ -0,0 +1,329 @@
+		extraData: '',
+		sourcePage: sourcepage,
+		viewName,
+	});
+
+	const { mutateAsync: deleteViewAsync } = useDeleteView(viewKey);
+
+	const onDeleteHandler = useCallback(async () => {
+		deleteViewHandler({
+			deleteViewAsync,
+			notifications,
+			panelType,
+			redirectWithQueryBuilderData,
+			refetchAllView,
+			viewId: viewKey,
+			viewKey,
+		});
+	}, [
+		deleteViewAsync,
+		notifications,
+		panelType,
+		redirectWithQueryBuilderData,
+		refetchAllView,
+		viewKey,
+	]);
+
+	const onUpdateQueryHandler = useCallback(async () => {
+		await updateQueryHandler({
+			compositeQuery: mapCompositeQueryFromQuery(currentQuery, panelType),
+			viewKey,
+			extraData: '',
+			sourcePage: sourcepage,
+			viewName,
+			notifications,
+			setIsQueryUpdated,
+			updateViewAsync,
+		});
+		refetchAllView();
+	}, [
+		currentQuery,
+		notifications,
+		panelType,
+		refetchAllView,
+		sourcepage,
+		updateViewAsync,
+		viewKey,
+		viewName,
+	]);
+
+	const onMenuItemSelectHandler = useCallback(
+		({ key }: { key: string }): void => {
+			const currentViewDetails = getViewDetailsUsingViewKey(
+				key,
+				viewsData?.data?.data,
+			);
+			if (!currentViewDetails) return;
+			const { query, name, uuid } = currentViewDetails;
+
+			// AffregateOperator should be noop for list and trace Panel Type and count for graph and table Panel Type.
+			query.builder.queryData = query.builder.queryData.map((item) => {
+				const newItem = item;
+				if (currentPanelType === 'list' || currentPanelType === 'trace') {
+					newItem.aggregateOperator = 'noop';
+				} else {
+					newItem.aggregateOperator = 'count';
+				}
+				return newItem;
+			});
+
+			redirectWithQueryBuilderData(query, {
+				[queryParamNamesMap.panelTypes]: panelType,
+				[querySearchParams.viewName]: name,
+				[querySearchParams.viewKey]: uuid,
+			});
+		},
+		[
+			viewsData?.data?.data,
+			redirectWithQueryBuilderData,
+			panelType,
+			currentPanelType,
+		],
+	);
+
+	useEffect(() => {
+		if (copyUrl.value) {
+			notifications.success({
+				message: 'Copied to clipboard',
+			});
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [copyUrl.value]);
+
+	useEffect(() => {
+		setIsQueryUpdated(
+			isQueryUpdatedInView({
+				data: viewsData?.data?.data,
+				stagedQuery,
+				viewKey,
+			}),
+		);
+	}, [
+		currentQuery,
+		viewsData?.data?.data,
+		stagedQuery,
+		stagedQuery?.builder.queryData,
+		viewKey,
+	]);
+
+	const generatorMenuItems = useCallback(
+		(data: ViewProps[] | undefined): MenuProps['items'] => {
+			if (!data) return [];
+
+			return data.map((view) => ({
+				key: view.uuid,
+				label: (
+					<MenuItemGenerator
+						viewName={view.name}
+						viewKey={viewKey}
+						createdBy={view.createdBy}
+						uuid={view.uuid}
+						refetchAllView={refetchAllView}
+						onMenuItemSelectHandler={onMenuItemSelectHandler}
+					/>
+				),
+			}));
+		},
+		[onMenuItemSelectHandler, refetchAllView, viewKey],
+	);
+
+	const updateMenuList = useMemo(
```

Comment:
useCallback is wrapped under useMemo is not a good approach please optimise this function though we statically generating the menu item and attaching onClick handler

Reply:
okay, the onClick handler will not be applied here as we are using ReactNode as a  label with two components inside to switch the view and to delete the view.